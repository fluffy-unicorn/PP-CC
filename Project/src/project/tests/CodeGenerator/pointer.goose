int* i = 5;
char* c = 'n';
bool* b = false;
function void main() {
	print i;
	print c;
	print b;
	change(i);
	change(c);
	change(b);
	print i;
	print c;
	print b;
}

function void change(int* i) {
	i = 7;
}

function void change(char* c) {
	c = 'p';
}

function void change(bool*  b) {
	b = true;
}


