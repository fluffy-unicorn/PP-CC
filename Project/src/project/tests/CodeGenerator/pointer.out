Sprockell 0 says 5
Sprockell 0 says n
Sprockell 0 says False
Sprockell 0 says 7
Sprockell 0 says p
Sprockell 0 says True
